bluesky/protocols.py

```diff
@@ -227,14 +243,62 @@ def subscribe(self, function: Callable) -> Union[NoReturn, None]:
         """
         ...
 
-    def clear_sub(self, function: Callable) -> Union[NoReturn, None]:
+    def clear_sub(self, function: Callable) -> None:
         """Remove a subscription."""
         ...
 
 
+class Subscription(Protocol):
+    def __enter__(self) -> "Subscription":
+        """Returns self"""
+        return self
+
+    def __exit__(self, type, value, traceback):
+        """Calls self.close()"""
+        self.close()
+
+    def close(self):
+        """Closes connection"""
+        ...
+
+
+@runtime_checkable
+class Monitorable(Protocol):
+    """Async compatible version of Subscribable with tweaked interface"""
+    # TODO: monitor() returns a Subscription, do we like the naming?
```

Comment:
I had a name collision here, so I renamed the async version of Subsribable "Monitorable" after the bluesky verb... Do we like the name? And the interface?

Reply:
Not sure what the best way forward is here but if we're going to have separate sync and async protocols for something I think there should be a deprecation procedure or at least some `TODO` about future reconciliation.